hn/client.go

```diff
@@ -0,0 +1,61 @@
+package hn
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+type Client struct {
+	baseUrl string
+}
+
+type ClientOption func(c *Client)
+
+func WithBaseUrl(baseUrl string) ClientOption {
+	return func(c *Client) {
+		c.baseUrl = baseUrl
+	}
+}
+
+func New(opts ...ClientOption) *Client {
+	c := &Client{
```

Comment:
I would probably add http client onto your Client, so you can replace it in testing??

Reply:
Good idea 👍